src/Microsoft.HttpRepl/Commands/BaseHttpCommand.cs

```diff
@@ -118,175 +113,159 @@ protected override async Task ExecuteAsync(IShellState shellState, HttpState pro
-            ResponseConfig responseConfig = new ResponseConfig(_preferences);
             string protocolInfo;
 
             if (echoRequest)
             {
-                string hostString = response.RequestMessage.RequestUri.Scheme + "://" + response.RequestMessage.RequestUri.Host + (!response.RequestMessage.RequestUri.IsDefaultPort ? ":" + response.RequestMessage.RequestUri.Port : "");
-                consoleManager.WriteLine($"Request to {hostString}...".SetColor(requestConfig.AddressColor));
-                consoleManager.WriteLine();
-
-                string method = response.RequestMessage.Method.ToString().ToUpperInvariant().SetColor(requestConfig.MethodColor);
-                string pathAndQuery = response.RequestMessage.RequestUri.PathAndQuery.SetColor(requestConfig.AddressColor);
-                protocolInfo = $"{"HTTP".SetColor(requestConfig.ProtocolNameColor)}{"/".SetColor(requestConfig.ProtocolSeparatorColor)}{response.Version.ToString().SetColor(requestConfig.ProtocolVersionColor)}";
-
-                consoleManager.WriteLine($"{method} {pathAndQuery} {protocolInfo}");
-                IEnumerable<KeyValuePair<string, IEnumerable<string>>> requestHeaders = response.RequestMessage.Headers;
-
-                if (response.RequestMessage.Content != null)
-                {
-                    requestHeaders = requestHeaders.Union(response.RequestMessage.Content.Headers);
-                }
-
-                foreach (KeyValuePair<string, IEnumerable<string>> header in requestHeaders.OrderBy(x => x.Key))
-                {
-                    string headerKey = header.Key.SetColor(requestConfig.HeaderKeyColor);
-                    string headerSep = ":".SetColor(requestConfig.HeaderSeparatorColor);
-                    string headerValue = string.Join(";".SetColor(requestConfig.HeaderValueSeparatorColor), header.Value.Select(x => x.Trim().SetColor(requestConfig.HeaderValueColor)));
-                    consoleManager.WriteLine($"{headerKey}{headerSep} {headerValue}");
-                }
-
-                consoleManager.WriteLine();
-
-                List<string> responseOutput = new List<string>();
-
-                if (response.RequestMessage.Content != null)
-                {
-                    await FormatBodyAsync(commandInput, programState, consoleManager, response.RequestMessage.Content, responseOutput, _preferences, cancellationToken).ConfigureAwait(false);
-                }
-
-                consoleManager.WriteLine();
-                consoleManager.WriteLine($"Response from {hostString}...".SetColor(requestConfig.AddressColor));
-                consoleManager.WriteLine();
-                foreach (string responseLine in responseOutput)
-                {
-                    consoleManager.WriteLine(responseLine);
-                }
+                protocolInfo = await HandleEchoRequest(commandInput, consoleManager, programState, response, cancellationToken);
```

Comment:
Something doesn't seem right here. `protocolInfo` is set here, but then will be almost immediately overwritten on L269. It looks like it is used in `HandleEchoRequest`. Does it just not need to be returned?

Reply:
Ah, actually we don't need to return it. It just needs to be written to console manager. Updated here:
https://github.com/aspnet/HttpRepl/commit/a41ab80846f7a8137b96f77c1e2a0b8cb2b83e34